src/Microsoft.Performance.SDK.Runtime/AssemblyTypeReference.cs

```diff
@@ -55,25 +64,64 @@ protected AssemblyTypeReference(AssemblyTypeReference<TDerived> other)
         {
             Debug.Assert(other != null);
 
-            this.Type = other.Type;
-            this.AssemblyPath = other.AssemblyPath;
-            this.Version = other.Version;
+            this.type = other.Type;
+            this.assemblyPath = other.AssemblyPath;
+            this.version = other.Version;
+        }
+
+        /// <summary>
+        ///     Finalizes an instance of the <see cref="AssemblyTypeReference{TDerived}"/>
+        ///     class.
+        /// </summary>
+        ~AssemblyTypeReference()
+        {
+            this.Dispose(false);
         }
```

Comment:
My understanding after reviewing CLR via C# is that adding a finalizer is intended for use when handling a native resource. Adding a finalizer means that the object has exists for an extra generation in the GC, as do any of its references. Since most of these classes don't directly own native resources, I don't believe that adding a finalizer helps, and it actually hurts.

Reply:
You're right, we don't need it.